test/auth.test.js

```diff
@@ -0,0 +1,190 @@
+                });
+            }); 
+    
+        });
+    
+    
+        describe('signing up with all the required fields', ()=>{
+            
+            it('should return a 201 status code has a data property with a token', (done)=>{
+                chai.request(server)
+                .post(SIGN_UP_ROUTE)
+                .send({
+                    user:{
+                        email: faker.internet.email(),
+                        username:faker.name.lastName() + Date.now(),
+                        password: faker.internet.password(),
+                    }
+                })
+                .end((err, res)=>{
+                    expect(res.status).to.equal(201 );
+                    expect(res.body).to.haveOwnProperty('data');
+                    expect(res.body.data).to.haveOwnProperty('token');
+                    done();
+                });
+            }); 
+    
+        });
+    });
+    
+    
+    
+    
+    describe('user on loging in ', ()=>{
+        before((done)=>{
+            chai.request(server)
+            .post(SIGN_UP_ROUTE)
+            .send(tempUser)
+            .end((err, res)=>{
+               done();
+            });
+        });
+       
+        describe('sloging in with no username', ()=>{
+            it('should not return  a data property', (done)=>{
+                chai.request(server)
+                .post(LOGIN_ROUTE)
+                .send({
+                    user:{
+                        password: faker.internet.password()
+                    }
+                })
+                .end((err, res)=>{
+                    expect(res.body)
+                       .to.not.have.property('data');
+                    done();
+                });
+            }); 
+    
+        });
+    
+        describe('loging in with no password', ()=>{
+            it('should not return a data property ', (done)=>{
+                chai.request(server)
+                .post(LOGIN_ROUTE)
+                .send({
+                    user:{
+                        username: faker.name.lastName() + Date.now(),
+                    }
+                })
+                .end((err, res)=>{
+                    expect(res.body)
+                       .to.not.have.property('data');
+                    done();
+                });
+            }); 
+    
+        });
+    
+    
+        
+        describe('loging in with all the required fields', ()=>{
+            it('should return a 200 status code has a data property with a token', (done)=>{
+    
+                
+                chai.request(server)
+                .post(LOGIN_ROUTE)
+                .send(tempUser)
+                .end((err, res)=>{
+                    expect(res.status).to.equal(200);
+                    expect(res.body).to.haveOwnProperty('data');
+                    expect(res.body.data).to.haveOwnProperty('token');
+                    done();
+                });
+            }); 
+    
+        });
+    });
```

Comment:
To fix your indentation issues, you can set your editor spaces to `2` at the bottom left corner of your VS code editor. Then press `option` + `shift` + `f` on the file to auto correct the indentation.